sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -211,50 +216,44 @@ public void AddTag(string name, object value)
                 }
             }
 
-            private List<ActivityLink>? GetActivitySourceLinkCollection()
+            private IReadOnlyList<Activity> GetDiagnosticSourceLinkCollection()
             {
                 if (_links == null)
                 {
-                    return null;
+                    return Array.Empty<Activity>();
                 }
 
-                var linkCollection = new List<ActivityLink>();
+                var linkCollection = new List<Activity>();
 
-                foreach (var activity in _links)
+                foreach (var link in _links)
                 {
-                    ActivityTagsCollection linkTagsCollection = new();
-                    foreach (var tag in activity.Tags)
+                    var activity = new Activity("LinkedActivity");
+                    if (link.Context != default) // TODO
                     {
-                        linkTagsCollection.Add(tag.Key, tag.Value!);
+                        string flags = (link.Context.TraceFlags == ActivityTraceFlags.None) ? "00" : "01";
+                        activity.SetParentId("00-" + link.Context.TraceId + "-" + link.Context.SpanId + "-" + flags);
                     }
+                    activity.TraceStateString = link.Context.TraceState;
 
-                    var contextWasParsed = ActivityContext.TryParse(activity.ParentId!, activity.TraceStateString, out var context);
-                    if (contextWasParsed)
+                    if (link.Tags != null)
                     {
-                        var link = new ActivityLink(context, linkTagsCollection);
-                        linkCollection.Add(link);
+                        foreach (var tag in link.Tags)
+                        {
+                            // old code path, only string attributes are supported
+                            activity.AddTag(tag.Key, tag.Value?.ToString());
```

Comment:
Do we expect `tag.Value` to sometimes be null here? If so, when it is null do we want to still add the tag?